app/uk/gov/hmrc/eusubsidycompliancefrontend/connectors/EscConnector.scala

```diff
@@ -161,14 +161,14 @@ class EscConnectorImpl @Inject()(http: HttpClient,
       update = UndertakingSubsidyAmendment(
         List(
           NonHmrcSubsidy(
-            subsidyUsageTransactionID = None,
+            subsidyUsageTransactionID = journey.existingTransactionId,
             allocationDate = currentDate,
             submissionDate= currentDate,
             publicAuthority = Some(journey.publicAuthority.value.getOrElse(sys.error(" publicAuthority is missing"))),// this shouldn't be optional, is required in create API but not retrieve
```

Comment:
minor: instead of sys.error we can use the function created by carwyn -> handleMissingSessionData, just for consistency. what do you think?

Reply:
Sounds good!

Reply:
Looks like that's only in the base controller. I'll throw an exception there, but also do validation at the controller level